web/client/epics/maps.js

```diff
@@ -210,17 +218,148 @@ const reloadMapsEpic = (action$, { getState = () => { } }) =>
-            return Rx.Observable.fromPromise(GeoStoreApi.getResourcesByCategory(map, searchText, opts)
-                .then(data => data))
-                .switchMap((response) => Rx.Observable.of(
-                    mapsLoaded(response, opts.params, searchText)
-                ))
-                .catch((e) => loadError(e));
+            const state = store.getState();
+            const searchFilter = searchFilterSelector(state) || {};
+            const userRole = userRoleSelector(state);
+            let {map, searchText, opts = {}} = action;
+            const searchFilterContexts = searchFilter.contexts && searchFilter.contexts.length > 0;
+            const actualSearchText = searchFilterContexts && searchText === '*' ? '' : searchText;
+            const makeFilter = () => ({
+                AND: {
+                    FIELD: [{
+                        field: ['NAME'],
+                        operator: ['ILIKE'],
+                        value: ['%' + actualSearchText + '%']
+                    }],
+                    OR: searchFilterContexts && {
+                        ATTRIBUTE: (searchFilter.contexts || []).map(context => ({
+                            name: ['context'],
+                            operator: ['EQUAL_TO'],
+                            type: ['STRING'],
+                            value: [context.id]
+                        }))
+                    }
+                }
+            });
+            const getContextNames = ({results, ...other}) => {
+                const maps = isArray(results) ? results : (results === "" ? [] : [results]);
+                return maps.length === 0 ?
+                    Rx.Observable.of({results, ...other}) :
+                    Rx.Observable.forkJoin(
+                        maps.map(({context}) => context ?
+                            getResource(context, {includeAttributes: false, withData: false, withPermissions: false})
+                                .switchMap(resource => Rx.Observable.of(resource.name)) :
+                            Rx.Observable.of(null))
+                    ).map(contextNames => ({
+                        results: zip(maps, contextNames).map(
+                            ([curMap, contextName]) => ({...curMap, contextName})),
+                        ...other
+                    }));
+            };
+
+            return (searchFilterContexts ? searchListByAttributes(makeFilter(), {
+                ...opts,
+                params: {
+                    ...(opts.params || {}),
+                    includeAttributes: true
+                }
+            })
+                .switchMap(({results, totalCount}) => {
+                    const maps = {
+                        results: results.map(result => ({
+                            ...omit(result, 'attributes', 'permissions'),
+                            ...pick(result.attributes, 'thumbnail', 'context'),
+                            canCopy: userRole === 'ADMIN',
```

Comment:
Correct me if I'm wrong. You added this to results because the remove service doesn't provide this information (canCopy, canDelete, canEdit), isn't it? 
We can not assume that edit/delete are allowed only to admin user. Maybe we need to extend the entry point.

Reply:
I've extended the searchListByAttributes handling function to handle option "withPermissions" which when true, gets permissions for every returned resource with additional getResourcePermissions calls. It returns read/write permission though. canDelete and canEdit could be set to true if the user can write and canCopy if the user can read or write. Should I do this?

Reply:
You should read the full permission list and guess if you have permission or not. I prefer if this information is provided by the server and anyway it requires a request for each result. I prefer if the server returns these information directly

Reply:
> I prefer if the server returns these information directly

That would be ideal, yes.